raft-tech/TANF-app
# Feat/217 revise stt model

## Summary of Changes
Pull request closes #217 

This pr make a couple small changes to the STT views in the DAC. The `code` field has been changed to `postal_code` for clarity. A `sample` field has been created and is populated from the csv files.

## How to Test
```
cd tdrs-frontend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
cd tdrs-backend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d 
```

1. Open http://localhost:3000/ and sign in.
2. Go to `/admin/stts/stt/` and you will see that a new field named `sample` has been added and populated, the previously named `code` field has been changed to `postal_code`, and the `state` field is no longer populated for states or territories, only tribes.
3. The same changes as above will also be shown in the individual STT view page.
![Screen Shot 2023-01-18 at 10 47 22 AM](https://user-images.githubusercontent.com/7119690/213268029-6171bcb5-fce5-4bcc-a05a-8fe1b283e368.png)
![Screen Shot 2023-01-18 at 10 33 17 AM](https://user-images.githubusercontent.com/7119690/213268289-dc4f2ae5-3c6b-456e-bcae-6b5283b5b0a9.png)
![Screen Shot 2023-01-18 at 10 33 27 AM](https://user-images.githubusercontent.com/7119690/213268293-be95c985-bca7-4492-beea-835901eba68f.png)


## Deliverables

### [Deliverable 1: Accepted Features](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)


ACs: 
- [x] Type and State field need to interoperate. `State` field should be unavailable if type selected is state. 
- [x] all fields should be `read_only`

Tasks: 
- [x] Adjust `type` field for tribes and states (i.e., do not show the state field for a state object).
- [x] Remove duplicative `code_number` from stt model [here](https://github.com/raft-tech/TANF-app/blob/develop/tdrs-backend/tdpservice/stts/models.py#L31)
- [x] `Code` field should be renamed to represent that it is an abbreviation for the postal service. (e.g. `Postal`)
- [x] Add a `Sample` indicator (for a future ticket: if ==0, then any Section 4 data file should be rejected)

### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

No changes to tests 

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)


_n/a_ backend changes only

### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

_n/a_
_n/a_
